### DISCUSSION TITLE
Environment changed to lower case without being mentioned as breaking change?

### DISCUSSION THREAD

**Discussion by genne** [getsentry/sentry-dotnet#899] [Q&A] [answered]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
Sentry.AspNetCore 3.0.3

### Steps to Reproduce

1. Upgraded from 2.1.5 to 3.0.3
2. All existing alert rules for our production environment stopped working, since now the environment was "production" instead of "Production"

### Expected Result

I would expect it to keep the original behavior (i.e. keep "Production"), or at least be mentioned as a breaking change.

### Actual Result

We didn't get notified on production issues for over a week, and it took some time to figure out what happened.

**Comment by Tyrrrz** [comment]
It was mentioned in releases: https://github.com/getsentry/sentry-dotnet/releases/tag/3.0.0

And also in docs: https://docs.sentry.io/platforms/dotnet/migration/

It was not mentioned in `Changelog.md` though, as it's basically just a glorified list of PRs without any additional information.

cc @bruno-garcia

**Comment by bruno-garcia** [comment, marked as answer]
Sorry this cause some friction on your alerts. Sentry defaulted in all sdks to `production` and for ASP.NET Core that has the same behavior but with `Production` this became a problem and we had to introduce this breakage at somepoint. Since 3.0.0 was landing we decided it was the right time.
If you have any other Sentry SDK like JavaScript you'd end up with `Production` and `production` on the list forever. So we decided to set to `production` if it wasn't an explicitly set environment, but picked up from ASP.NET Core instead.

See: https://github.com/getsentry/sentry-dotnet/pull/550/files
And: https://github.com/getsentry/sentry-dotnet/pull/554/files

You can keep `Production` by setting it explicitly via the SentryOptions.Environment property:

`UseSentry(o => o.Environment = "Production"` in which case the SDK will take it as-is.

**Comment by genne** [comment]
Ah ok so that's good, then the issue was more that we didn't know where to look. Since you mentioned breaking changes for version 2 in your changelog we assumed that there were no breaking changes in version 3. Perhaps you could link to the release tag from the changelog, and highlight that there's where you can read about breaking changes?

(Another question is if Sentry environments really need to be case sensitive?)
